support-workers/src/main/scala/com/gu/salesforce/SalesforceService.scala

```diff
@@ -51,15 +51,24 @@ class SalesforceService(config: SalesforceConfig, client: FutureHttpClient)(impl
       recipientContactRecord <- maybeAddGiftRecipient(
         contactRecord.ContactRecord,
         giftRecipient,
-        user.deliveryAddress.getOrElse(user.billingAddress)
+        user
       )
     } yield SalesforceContactRecordsResponse(contactRecord, recipientContactRecord)
   }
 
-  private def maybeAddGiftRecipient(contactRecord: SalesforceContactRecord, maybeGiftRecipient: Option[GiftRecipient], deliveryAddress: Address) =
+  private def deliveryAddressForGiftRecipientType(giftRecipient: GiftRecipient, user: User) =
+    giftRecipient match {
+      case _: GiftRecipient.WeeklyGiftRecipient => user.deliveryAddress
+      case _: GiftRecipient.DigitalSubscriptionGiftRecipient  => None
+    }
+
+  private def maybeAddGiftRecipient(contactRecord: SalesforceContactRecord, maybeGiftRecipient: Option[GiftRecipient], user: User) =
     maybeGiftRecipient
       .filter(recipient => recipient.firstName != "" && recipient.lastName != "")
```

Comment:
existing code I know, but it seems a bit late to be checking for empty strings! I'm sure the validators check this also

Reply:
Yeah I was a bit confused about this as well, but as you say it was existing code so I left it.